flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/spec/ConfigObjectNode.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.api.spec;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/** */
+public class ConfigObjectNode extends ObjectNode {
+
+    public ConfigObjectNode() {
+        this(JsonNodeFactory.instance);
+    }
+
+    public ConfigObjectNode(JsonNodeFactory nc, Map<String, JsonNode> kids) {
+        super(nc, kids);
+    }
+
+    public ConfigObjectNode(JsonNodeFactory nc) {
+        super(nc);
+    }
+
+    public void removeAll(String... names) {
+        for (String name : names) {
+            remove(name);
+        }
+    }
+
+    public void putAllFrom(Map<String, String> value) {
+        value.forEach(this::put);
+    }
+
+    public void set(Map<String, String> value) {
+        removeAll();
```

Comment:
This looks a bit confusing, we probably shouldn't introduce a varargs variant of the same method that does something completely different :)

Reply:
changed it to `public void setAllFrom(Map<String, String> value)`